### ISSUE TITLE
MD052 Breaks with Hugo parameters in the URLs

### ISSUE THREAD

**Issue by ivanvc** [DavidAnson/markdownlint#1619]
## What did you do?

I have the following markdown file (`test.md`):

```markdown
# Test

Test [link to somewhere][].

[link to somewhere]: /docs/{{<param version >}}/test
```

## What did you expect to happen?

Running `markdownlint-cli2` should not raise an MD052 issue.

## What actually happened

It fails to run:

```
$ docker run -w /src -v $PWD:/src --rm -ti davidanson/markdownlint-cli2 test.md
markdownlint-cli2 v0.18.1 (markdownlint v0.38.0)
Finding: test.md
Linting: 1 file(s)
Summary: 2 error(s)
test.md:3:6 MD052/reference-links-images Reference links and images should use a label that is defined [Missing link or image reference definition: "link to somewhere"] [Context: "[link to somewhere][]"]
test.md:5:30 MD033/no-inline-html Inline HTML [Element: param]
```

## What messages or errors were there?

```
test.md:3:6 MD052/reference-links-images Reference links and images should use a label that is defined [Missing link or image reference definition: "link to somewhere"] [Context: "[link to somewhere][]"]
```

## How can the issue be reproduced?

Use the reference markdown file, and run `docker run -w /src -v $PWD:/src --rm -ti davidanson/markdownlint-cli2 test.md`

## What version were you using?

```
markdownlint-cli2 v0.18.1 (markdownlint v0.38.0)
```

## What operating system were you using?

Reproducible with Docker, the host machine is running Linux.

**Comment by ivanvc** [comment]
Thanks for the answer, @DavidAnson. I took a look and found that because we're using Docsy (which utilizes [Hugo inline shortcodes](https://gohugo.io/content-management/shortcodes/#inline)) to generate the links, the URLs don't adhere to the CommonMark standard.

Closing.
